stingray/pulse/overlapandsave/ols.py

```diff
@@ -0,0 +1,355 @@
+"""
+The code in this file was copied from
+https://github.com/fasiha/array_range
+https://github.com/fasiha/nextprod-py
+https://github.com/fasiha/overlap_save-py
+
+Thanks to Ahmed Fasih for releasing this to the larger public with the
+Unlicense.
```

Comment:
Basically, you can do whatever you want with the code, it's even more permissive than BSD. I did want to acknowledge the author though, and I'm backporting to the original library all the fixes/extensions I find useful.
I wonder if we can just make a fork of that code under stingraysoftware to adapt it to our needs (and use our fork in stingray), instead of putting it in stingray, since it's general purpose code that other people might find useful besides stingray. I don't want to add it as an external/pip dependency, because the API might change.